nautobot/core/jobs/__init__.py

```diff
@@ -361,6 +371,131 @@ def run(self, *, content_type, csv_data=None, csv_file=None, roll_back_if_error=
+        label="Select Data Compliance Rules",
+        required=False,
+        description="Not selecting any rules will run all rules listed.",
+    )
+
+    run_builtin_rules_in_report = BooleanVar(
+        label="Run built-in validation rules?", description="Include created, built-in data validation rules in report."
+    )
+
+    def run(self, *args, **kwargs):
+        """Run the validate function on all given DataComplianceRule classes."""
+        selected_data_compliance_rules = kwargs.get("selected_data_compliance_rules", None)
+
+        compliance_classes = []
+        compliance_classes.extend(get_data_compliance_rules())
+
+        for repo in GitRepository.objects.all():
+            if "nautobot_data_validation_engine.data_compliance_rules" in repo.provided_contents:
+                compliance_classes.extend(get_classes_from_git_repo(repo))
+
+        for compliance_class in compliance_classes:
+            if selected_data_compliance_rules and compliance_class.__name__ not in selected_data_compliance_rules:
+                continue
+            self.logger.info(f"Running {compliance_class.__name__}")
+            app_label, model = compliance_class.model.split(".")
+            for obj in global_apps.get_model(app_label, model).objects.all():
+                ins = compliance_class(obj)
+                ins.enforce = False
+                ins.clean()
+
+        run_builtin_rules_in_report = kwargs.get("run_builtin_rules_in_report", False)
+        if run_builtin_rules_in_report:
+            self.logger.info("Running built-in data validation rules")
+            self.report_for_validation_rules()
+
+        self.logger.info("View Data Compliance results [here](/data-validation-engine/data-compliance/)")
+
+    @staticmethod
+    def report_for_validation_rules():
+        """Run built-in data validation rules and add to report."""
+        query = (
+            Q(uniquevalidationrule__isnull=False)  # pylint: disable=unsupported-binary-operation
+            | Q(regularexpressionvalidationrule__isnull=False)
+            | Q(minmaxvalidationrule__isnull=False)
+            | Q(requiredvalidationrule__isnull=False)
+        )
+
+        model_classes = [ct.model_class() for ct in ContentType.objects.filter(query).distinct()]
+
+        # Gather custom validators of built-in rules
+        validator_dicts = []
+        for model_class in model_classes:
+            model_custom_validators = registry["plugin_custom_validators"][model_class._meta.label_lower]
+            # Get only DataValidationCustomValidators
+            # otherwise, we would get all validators (more than those dynamically created)
+            validator_dicts.extend(
+                [
+                    {cv: model_class}
+                    for cv in model_custom_validators
+                    if cv.__name__
+                    == f"{model_class._meta.app_label.capitalize()}{model_class._meta.model_name.capitalize()}CustomValidator"
```

Comment:
This seems really fragile to me, maybe there's a better way we can do this, like a class attribute that specifies it's dynamically created?

Reply:
I think this is the corresponding [code](https://github.com/nautobot/nautobot/blob/next/nautobot/extras/plugins/validators.py#L31-L35) that wraps the dynamically generated custom validators `clean` method to the `model.clean` method. I'm not sure why we are doing it this way.

Reply:
https://github.com/nautobot/nautobot/pull/7074/commits/c9e43ec8ad8b8b5d842315d5f946f20d9a018db5